app/frontend/packs/components/checkbox_search_filter.js

```diff
@@ -0,0 +1,202 @@
+    })
+  }
+
+  getAllCheckboxes () {
+    return this.checkboxesContainer.querySelectorAll('.govuk-checkboxes__item')
+  }
+
+  getAllVisibleCheckboxes () {
+    return [].filter.call(this.getAllCheckboxes(), function (el) {
+      return el.style.display === 'block'
+    })
+  }
+
+  getAllVisibleCheckedCheckboxes () {
+    return [].filter.call(this.getAllVisibleCheckboxes(), function (el) {
+      return el.querySelector('.govuk-checkboxes__input').checked
+    })
+  }
+
+  getTextBoxHtml () {
+    const containerId = this.container.id
+    const label = document.createElement('label')
+    label.setAttribute('for', containerId + '-checkbox-filter__filter-input')
+    label.className = 'govuk-label govuk-visually-hidden'
+    label.innerHTML = this.params.textBox.label
+
+    const input = document.createElement('input')
+    input.className = 'app-checkbox-filter__filter-input govuk-input'
+    input.setAttribute('id', containerId + '-checkbox-filter__filter-input')
+    input.setAttribute('type', 'text')
+    input.setAttribute('aria-describedby', containerId + '-checkboxes-status')
+    input.setAttribute('aria-controls', containerId + '-checkboxes')
+    input.setAttribute('autocomplete', 'off')
+    input.setAttribute('spellcheck', 'false')
+
+    return [label, input]
+  }
+
+  getVisibleCheckboxes () {
+    const visibleCheckboxes = [].filter.call(this.checkboxes, (el) => this.isCheckboxInView(el))
+
+    // add an extra checkbox, if the label of the first is too long it collapses onto itself
+    visibleCheckboxes.push(this.checkboxes[visibleCheckboxes.length - 1])
+    return visibleCheckboxes
+  }
+
+  isCheckboxInView (checkbox) {
+    const initialOptionContainerHeight = this.checkboxesContainer.style.height
+    const optionListOffsetTop = this.checkboxesInnerContainer.offsetTop
+    const distanceFromTopOfContainer = checkbox.offsetTop - optionListOffsetTop
+    return distanceFromTopOfContainer < initialOptionContainerHeight
+  }
+
+  onTextBoxKeyUp (e) {
+    const ENTER_KEY = 13
+    if (e.keyCode === ENTER_KEY) {
+      e.preventDefault()
+    } else {
+      this.filterCheckboxes()
+    }
+  }
+
+  sanitizeText (text) {
+    text = text.replace(/&/g, 'and')
+    text = text.replace(/[’',:–-]/g, '') // remove punctuation characters
+    text = text.replace(/[.*+?^${}()|[\]\\]/g, '\\$&') // escape special characters
+    return text.trim().replace(/\s\s+/g, ' ').toLowerCase() // replace multiple spaces with one
+  }
+
+  setContainerHeight (height) {
+    this.checkboxesContainer.style.height = height
+  }
+
+  setup () {
+    this.setupStatusBox()
+    this.setupHeading()
+    if (this.checkboxes.length >= 15) {
+      this.setupTextBox()
+    }
+    this.setupHeight()
+  }
+
+  setupHeading () {
+    const element = document.createElement('p')
+    element.className = 'app-checkbox-filter__title'
+    element.setAttribute('aria-hidden', 'true')
+
+    this.heading = element
+    this.container.prepend(this.heading)
```

Comment:
Does this get polyfilled? It may break some browsers if it doesn't
https://caniuse.com/?search=prepend

Reply:
workaround `this.container.insertBefore(this.heading, this.container.firstChild)`